tests/user_input_checks_test.py

```diff
@@ -435,3 +436,43 @@ def test_passing_custom_density_estimator(arg):
         density_estimator = arg
     prior = MultivariateNormal(torch.zeros(2), torch.eye(2))
     _ = SNPE_C(prior=prior, density_estimator=density_estimator)
+
+
```

Comment:
Please add:
```
@pytest.mark.gpu
```

Reply:
I can add that, but would that skip/exclude the unit test **completely** if I am running on a cpu-only host? if so, then I'd rather split the unit test.

Reply:
I will skip the test only if you specify `pytest -m 'not gpu' tests/`.

Reply:
Ok, then let me split the unit test accordingly so that this unit test mark can be added.

Reply:
done with 177f942